### PR TITLE
restructure files

### PR DESCRIPTION
**Pull request by notrab**
The big refactor...

Not in this PR

* `website` rename
* `README` move (will move once @FrancoAguzzi merges his changes and our PRs settle this week)

### CODE REVIEW

**Review by lightwalker-eth — CHANGES_REQUESTED** [review]
@notrab Hey from a code-review perspective looks good. Before we merge this though appreciate if you could action the following:

1. I see there's a build error for the NameGuard website. https://vercel.com/namehash/nameguard-website/Et8sDahPi1r5FwnUwvweDPgW48LY
2. Create a PR on the readme in [ens-utils](https://github.com/namehash/ens-utils) so that the readme is basically just a link to where the new home for that library has moved.
3. Same for the [namekit-archive ](https://github.com/namehash/namekit-archive) repo. If you could update the readme there to basically just document the mappings from code in that repo to code in the new namekit monorepo.

Nice work :+1:

File: README.md
Comment (lightwalker-eth) [inline README.md:1]: @notrab Let's keep this file. I'm not sure of the best answer for what to do with it at the moment. I think these ideas are probably best transitioned into the mintlify docs for NameGuard later but looking for a quick answer for now.

This readme is a general readme for NameGuard overall. So I don't believe it should replace the readme for any of the separate NameGuard packages or apps.

A temporary suggestion for now is to keep this file but rename it to "nameguard.md". What do you think?

File: coverage_badge.svg
Comment (lightwalker-eth) [inline coverage_badge.svg:1]: Let's confirm some details about this file. Please see my other comment about this file for more info.

File: packages/nameguard-sdk/examples/with-nextjs/src/app/page.tsx
Comment (lightwalker-eth) [inline packages/nameguard-sdk/examples/with-nextjs/src/app/page.tsx:1]: Can you please confirm the plan for how this example app will (continue to) be automatically deployed to Vercel? Let's ensure the CI/CD works well for this example app. This will be important for some of the next steps that Jakub is advancing.

I'm fine if this is something we need to follow up with after we merge this PR. If this is true, can you please make a note to be sure we follow up with this? Appreciate your advice.

CleanShot 2024-05-15 at 11 35 06

File: README.md
Comment (lightwalker-eth) [inline README.md:4]: @djstrong @Carbon225 @notrab Appreciate your help to confirm what we should do with this coverage badge.

One of the confusing things about this badge for me is: Exactly what within this monorepo is this coverage related to? There's so many different sub-projects.

I see that the last time this badge was automatically updated was 4 months ago by a GitHub Action: https://github.com/namehash/nameguard/commits/dd6f9af2587348a4808cf15356338c10181d00ec/coverage_badge.svg

Is my memory on this correct? Did we temporarily add a GitHub Action for this badge and then later remove the GitHub Action but never cleaned up this badge properly?

I still see a GitHub Action that generates a coverage badge, but perhaps this is only for the `/api` project? https://github.com/namehash/nameguard/blob/dd6f9af2587348a4808cf15356338c10181d00ec/.github/workflows/ci_api.yml#L82-L84

This is confusing because when I check https://github.com/namehash/nameguard/tree/main/api I don't see any coverage badge there.

Is the solution to remove the coverage badge from the root of our monorepo and move it into /api with a reference to it inside /api/README.md? This might involve some updates to the related GitHub action?

Appreciate your advice. Thank you.

File: README.md
Comment (lightwalker-eth) [inline README.md:1]: @notrab As a temporary step in the new root-level NameKit readme, could you please include a link to the retained / renamed "nameguard.md"? The goal of this suggestion is to retain some better discoverability for where to learn more about getting started with NameGuard. Appreciate your advice.

**Review by lightwalker-eth — CHANGES_REQUESTED** [review]
@notrab Thanks for updates! Reviewed and shared feedback :+1:

File: packages/nameguard-sdk/examples/with-nextjs/src/app/page.tsx
Comment (notrab) [reply packages/nameguard-sdk/examples/with-nextjs/src/app/page.tsx:1]: I don't think this was ever automatically deployed due to some restrictions of Vercel, but I will double check.

File: README.md
Comment (notrab) [reply README.md:4]: This badge was related to the API, so this should really be placed inside the `api` folder and sourced from the README accordingly.

File: README.md
Comment (notrab) [reply README.md:1]: I planned to have a list of all the packages, internal packages and apps with links to their folders. I'll do that probably later today.

File: README.md
Comment (djstrong) [reply README.md:4]: Yes, it still working and relates to Python API. In https://github.com/namehash/nameguard/pull/261 I moved the badge to `api` directory.
It was updated 4 months ago because the coverage has not changed since then.

Probably there is problem with the first badge because now API tests are not run every time.

File: README.md
Comment (lightwalker-eth) [reply README.md:1]: @notrab I took care of this in this commit: https://github.com/namehash/nameguard/pull/260/commits/9514aa7bac9474a78d2288e971621077013d2f26

File: README.md
Comment (lightwalker-eth) [reply README.md:1]: @notrab Ok sounds awesome :+1:

I made a temporary solution in this commit: https://github.com/namehash/nameguard/pull/260/commits/9514aa7bac9474a78d2288e971621077013d2f26

Comment (notrab) [reply README.md:1]: Awesome, thank you @lightwalker-eth

**Review by lightwalker-eth — APPROVED** [review]
Approved for merge :+1:

**Pull request merged**
